syntax = "proto3";

package greet;

service Greeter {
  rpc SayHello(HelloRequest) returns (HelloReply) {}

  rpc ParrotSayHello(HelloRequest) returns (stream HelloReply) {}

  rpc ChattyClientSaysHello(stream HelloRequest) returns (DelayedReply) {}

  rpc InteractingHello(stream HelloRequest) returns (stream HelloReply) {}
}

message HelloRequest {
  string salutation = 1;
}

message HelloReply {
  string retort = 1;
}

message DelayedReply {
  string message = 1;
  repeated HelloRequest request = 2;
}
